internal/cmd/billing/command_cost_list.go

```diff
@@ -0,0 +1,107 @@
+package billing
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	pcmd "github.com/confluentinc/cli/internal/pkg/cmd"
+	"github.com/confluentinc/cli/internal/pkg/output"
+)
+
+type command struct {
+	*pcmd.AuthenticatedCLICommand
+}
+
+type costOut struct {
+	StartDate           string `human:"Start Date" serialized:"start_date"`
+	EndDate             string `human:"End Date" serialized:"end_date"`
+	Granularity         string `human:"Granularity" serialized:"granularity"`
+	LineType            string `human:"Line Type" serialized:"line_type"`
+	Product             string `human:"Product" serialized:"product"`
+	ResourceId          string `human:"Resource ID" serialized:"resource_id"`
+	ResourceDisplayName string `human:"Resource Display Name" serialized:"resource_display_name"`
+	EnvironmentId       string `human:"Environment ID" serialized:"environment_id"`
+	NetworkAccessType   string `human:"Network Access Type" serialized:"network_access_type"`
+	Price               string `human:"Price" serialized:"price"`
+	Unit                string `human:"Unit" serialized:"unit"`
+	OriginalAmount      string `human:"Original Amount" serialized:"original_amount"`
+	DiscountAmount      string `human:"Discount Amount" serialized:"discount_amount"`
+	Amount              string `human:"Amount" serialized:"amount"`
+}
+
+func (c *command) newListCostCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "list <start-date> <end-date>",
+		Example: "list 2023-01-01 2023-01-10",
+		Short:   "List Confluent Cloud billing costs.",
+		Long:    "List Confluent Cloud daily aggregated costs for a specific range of dates.",
+		Args:    cobra.ExactArgs(2),
+		RunE:    c.list,
+	}
+
+	pcmd.AddOutputFlag(cmd)
+
+	return cmd
+}
+
+func (c *command) list(cmd *cobra.Command, args []string) error {
+	startDate := args[0]
+	endDate := args[1]
+
+	_, err := time.Parse("2006-01-02", startDate)
+	if err != nil {
+		return fmt.Errorf("invalid start date format, expected format should look like: 2022-01-01")
+	}
+
+	_, err = time.Parse("2006-01-02", endDate)
+	if err != nil {
+		return fmt.Errorf("invalid end date format, expected format should look like: 2022-01-01")
+	}
+
+	costs, err := c.V2Client.ListBillingCosts(startDate, endDate)
+	if err != nil {
+		return err
+	}
+
+	list := output.NewList(cmd)
+	for _, cost := range costs {
+		out := &costOut{
+			StartDate:         cost.GetStartDate(),
+			EndDate:           cost.GetEndDate(),
+			Granularity:       cost.GetGranularity(),
+			LineType:          cost.GetLineType(),
+			Product:           cost.GetProduct(),
+			NetworkAccessType: cost.GetNetworkAccessType(),
+			Unit:              cost.GetUnit(),
+		}
+
+		// These fields may be empty depending on the line type, so casting floats as strings as to avoid zero-value
```

Comment:
The downside of doing this is these values will have quotes around them for JSON/YAML output. Can you copy what we did in `confluent price list`? (We have separate `humanOut` and `serializedOut` structs)

Reply:
For serializedOut if we keep using a `float64`, would the problem still exist where the price is 0? My initial thought was that if the field was a pointer `*float64` and it was nil, it would be either excluded from JSON/YAML or empty in human form, but it seems like using pointer will just show the pointer value.

Reply:
If the price is 0, is it not visible when it should be?

Reply:
it's not visible for line types that price doesn't make sense for. For example Promo code amounts do not have unit prices, similarly Support costs do not have unit price since the amount is usage depedendent.

Reply:
Makes sense, I don't think we've ever run in to this use case before. A temporary solution might be to create an `OptionalFloat64` type that overrides the `String()` function... let me think about this today and see if I can come up with a better solution.